src/types.js

```diff
@@ -113,3 +113,15 @@ export function isEnumValue(enumObj, s) {
   }
   return false;
 }
+
+/**
+ * Checks if the given key is a property in the map.
+ *
+ * @param {T}  obj a map like property.
+ * @param {string}  key
+ * @return {boolean}
+ * @template T
+ */
+export function hasOwn(obj, key) {
+  return Object.prototype.hasOwnProperty.call(obj, key);
```

Comment:
Nit: We can hoist `Object.prototype.hasOwnProperty`. https://github.com/ampproject/amphtml/pull/6592/files#diff-3a699a0719518f967e9b2df3f5549e02R18